service/kernel/data_management/data_conversion.cpp

```diff
@@ -31,115 +32,38 @@ namespace data_management
 {
 namespace internal
 {
-template <typename T1, typename T2>
-static void vectorConvertFunc(size_t n, const void * src, void * dst)
+
+template<typename T1, typename T2>
+static void vectorConvertFunc(size_t n, const void *src, void *dst)
 {
-    typedef void (*funcType)(size_t n, const void * src, void * dst);
-    static funcType ptr = 0;
-
-    if (!ptr)
-    {
-        int cpuid = (int)daal::services::Environment::getInstance()->getCpuId();
-
-        switch (cpuid)
-        {
-#ifdef DAAL_KERNEL_AVX512
-        case avx512: DAAL_KERNEL_AVX512_ONLY_CODE(ptr = vectorConvertFuncCpu<T1, T2, avx512>); break;
-#endif
-#ifdef DAAL_KERNEL_AVX512_MIC
-        case avx512_mic: DAAL_KERNEL_AVX512_MIC_ONLY_CODE(ptr = vectorConvertFuncCpu<T1, T2, avx512_mic>); break;
-#endif
-#ifdef DAAL_KERNEL_AVX2
-        case avx2: DAAL_KERNEL_AVX2_ONLY_CODE(ptr = vectorConvertFuncCpu<T1, T2, avx2>); break;
-#endif
-#ifdef DAAL_KERNEL_AVX
-        case avx: DAAL_KERNEL_AVX_ONLY_CODE(ptr = vectorConvertFuncCpu<T1, T2, avx>); break;
-#endif
-#ifdef DAAL_KERNEL_SSE42
-        case sse42: DAAL_KERNEL_SSE42_ONLY_CODE(ptr = vectorConvertFuncCpu<T1, T2, sse42>); break;
-#endif
-#ifdef DAAL_KERNEL_SSSE3
-        case ssse3: DAAL_KERNEL_SSSE3_ONLY_CODE(ptr = vectorConvertFuncCpu<T1, T2, ssse3>); break;
-#endif
-        default: ptr = vectorConvertFuncCpu<T1, T2, sse2>; break;
-        };
-    }
+    #define DAAL_FUNC_CPU(cpuId, ...) \
```

Comment:
I recommend to name macros more specifically to avoid name collision, to me `DAAL_FUNC_CPU` is too general, `DAAL_VECTOR_CONVERT_CPU` is better.